napari/_qt/widgets/qt_eliding_label.py

```diff
@@ -9,19 +9,23 @@ class ElidingLabel(QLabel):
     def __init__(self, text='', parent=None):
         super().__init__(parent)
         self._text = text
-        self.fm = QFontMetrics(self.font())
 
     def setText(self, txt):
         self._text = txt
-        short = self.fm.elidedText(self._text, Qt.ElideRight, self.width())
+        short = self._getShortText(self.width())
         super().setText(short)
 
     def resizeEvent(self, rEvent):
         width = rEvent.size().width()
-        short = self.fm.elidedText(self._text, Qt.ElideRight, width)
+        short = self._getShortText(width)
         super().setText(short)
         rEvent.accept()
 
+    def _getShortText(self, width):
+        self.fm = QFontMetrics(self.font())
```

Comment:
oOut of curiosity why was this moved out of init / in which condition does it become stale ?

Reply:
yeah, see https://github.com/napari/napari/pull/2255#issuecomment-779215124 ... she found that if she changed the font size after creating the label, the metric was incorrect